test/e2e_node/hugepages_test.go

```diff
@@ -38,19 +37,8 @@ import (
 	. "github.com/onsi/gomega"
 )
 
-// makePodToVerifyHugePages returns a pod that verifies specified cgroup with hugetlb
-func makePodToVerifyHugePages(baseName string, hugePagesLimit resource.Quantity) *apiv1.Pod {
-	// convert the cgroup name to its literal form
-	cgroupFsName := ""
-	cgroupName := cm.NewCgroupName(cm.RootCgroupName, defaultNodeAllocatableCgroup, baseName)
-	if framework.TestContext.KubeletConfig.CgroupDriver == "systemd" {
-		cgroupFsName = cgroupName.ToSystemd()
-	} else {
-		cgroupFsName = cgroupName.ToCgroupfs()
-	}
-
-	// this command takes the expected value and compares it against the actual value for the pod cgroup hugetlb.2MB.limit_in_bytes
-	command := fmt.Sprintf("expected=%v; actual=$(cat /tmp/hugetlb/%v/hugetlb.2MB.limit_in_bytes); if [ \"$expected\" -ne \"$actual\" ]; then exit 1; fi; ", hugePagesLimit.Value(), cgroupFsName)
+// makeHugePagePod returns a pod that requests the the given amount of huge page memory, and execute the given command
+func makeHugePagePod(baseName string, command string, totalHugePageMemory resource.Quantity, hugePageSize resource.Quantity) *apiv1.Pod {
```

Comment:
I suspect the change made here was to handle environments where 2MB pages were not available?  Can we still ensure that the hugetlb limit files using the cgroupfs are set as expected for the environment that the test is run against?  If I am following this change, we do not actually validate the pod cgroup setting is actually applied.

Reply:
Ish. The tests checks that you can allocate the requested amount, both using the mounted volume, and doing it "manually" in mmap. It also checks that allocation more than the requested amount will fail, and that allocation with another huge page size (same amount of memory) also will crash